ripc-protocol-tcp/src/main/java/io/ripc/protocol/tcp/TcpInterceptor.java

```diff
@@ -0,0 +1,7 @@
+package io.ripc.protocol.tcp;
+
+public interface TcpInterceptor<I, O, II, OO> {
```

Comment:
This shouldn't have 4 generics placeholders. 2 is sufficient with 2 transformative ones on the method. Using this in practice will be horrific if the names are of any length.